### PR TITLE
[m365_defender] Add support for Streaming Event types

### PR DESCRIPTION
**Pull request by vinit-elastic**
## What does this PR do?

- Added a data stream.
- Added data collection logic to the data stream.
- Added the ingest pipeline for the data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboards and visualizations.
- Added test for pipeline for the data stream.
- Added system test cases for the data stream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.16.0 || ^8.0.0`).

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/m365_defender directory.
- Run the following command to run tests. 
> elastic-package test 

## Screenshots

image (5)
image (4)
image (3)
image (2)
image (1)
image

### CODE REVIEW

File: packages/m365_defender/data_stream/event/_dev/test/pipeline/test-alert.log-expected.json [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                    },
+                    "entity_type": "File",
+                    "evidence": {
+                        "role": "Related"
+                    },
+                    "file": {
+                        "name": "ai_alternate_stream_s7f_48qa.txt:ai-8xNfG12345",
+                        "size": 27436
+                    },
+                    "folder_path": "C:\\\\Windows\\\\Temp",
```
Comment (efd6) [inline packages/m365_defender/data_stream/event/_dev/test/pipeline/test-alert.log-expected.json:54]: Out of curiosity, why are these coming in with doubled up backslash escape sequences? To escape the real path, `C:\Windows\Temp`, this should be `"C:\\Windows\\Temp"`. Is this just weirdness in the API?

Comment (vinit-elastic) [reply packages/m365_defender/data_stream/event/_dev/test/pipeline/test-alert.log-expected.json:54]: Yes, the API provided `\\`.

Comment (efd6) [reply packages/m365_defender/data_stream/event/_dev/test/pipeline/test-alert.log-expected.json]: That's unfortunate, and an excellent bit of historical commentary on why choosing `\` as a path separator may not have been the best idea in the world.

File: packages/m365_defender/data_stream/event/elasticsearch/ingest_pipeline/pipeline_alert.yml [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+      on_failure:
+        - append:
+            field: error.message
+            value: '{{{_ingest.on_failure_message}}}'
+  - rename:
+      field: json.properties.AttackTechniques
+      target_field: m365_defender.event.attack_techniques
+      ignore_missing: true
+  - script:
+        description: Extract threat subtechnique id & name.
```
Comment (efd6) [inline packages/m365_defender/data_stream/event/elasticsearch/ingest_pipeline/pipeline_alert.yml:335]: These are going to need to have their order canonicalised to not be brittle in testing.

**Review by efd6 — COMMENTED** [review]
We'll need to have the `technique.subtechnique` arrays canonicalised since map iteration order is not guaranteed to be consistent.

File: packages/m365_defender/data_stream/event/_dev/test/pipeline/test-alert.log-expected.json (hunk 1 above)
Comment (vinit-elastic) [reply packages/m365_defender/data_stream/event/_dev/test/pipeline/test-alert.log-expected.json]: Yeah, I couldn't agree with you more.

**Review by efd6 — APPROVED** [review]
Sorry for the delay; LGTM

**Comment by vinit-elastic** [comment]
Hey @efd6 - Would you mind holding on to the merge for a bit? Jamie and I are to discuss a couple of approaches to disable the deprecated data stream by default. And I might add a commit along with this PR. 

Sorry for the inconvenience, I will let you know once we discuss the strategy. 😄

**Review by efd6 — APPROVED** [review]
Still LGTM

**Pull request merged**